src/cli.rs

```diff
@@ -0,0 +1,135 @@
+mod info;
+mod init;
+
+use std::env;
+use std::io;
+use std::path::PathBuf;
+use std::process;
+
+use clap::arg_enum;
+use structopt::StructOpt;
+
+use crate::errors::*;
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "TMSU",
+    about = "A tool for tagging your files and accessing them through a virtual filesystem"
+)]
+struct TmsuOptions {
+    // Externalize global options to a separate struct for convenience
+    #[structopt(flatten)]
+    global_opts: GlobalOptions,
+
+    #[structopt(subcommand)]
+    cmd: SubCommands,
+}
+
+#[derive(Debug, StructOpt)]
+pub struct GlobalOptions {
```

Comment:
I haven't implement `--verbose` yet. My plan is to leverage the logging infrastructure, and use this flag to increase the default level (warning -> info, info -> debug, etc).